src/Oasis/Signer.cpp

```diff
@@ -0,0 +1,65 @@
+// Copyright © 2017-2020 Trust Wallet.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#include <Oasis/Signer.h>
+
+#include "Signer.h"
+#include "Address.h"
+
+using namespace TW;
+using namespace TW::Oasis;
+
+
+Proto::SigningOutput Signer::sign(const Proto::SigningInput& input) noexcept {
+    auto output = Proto::SigningOutput();
+    auto signer = Signer(input);
+    auto encoded = signer.build();
+    output.set_encoded(encoded.data(), encoded.size());
+    return output;
+}
+
+Data Signer::build() const {
+    // Create empty address var and check if value we want to load is valid
+    Address address(input.transfer().to());
+
+    // Load value on that address var we create before
+    Address::decode(input.transfer().to(), address);
+
+    // Convert values from string to unsigned long
+    auto amount = strtoul(input.transfer().amount().c_str(), nullptr,10);
```

Comment:
`strtoul` return type is `unsigned long`, it may have overflow issue, you need to be able to handle `oasis total supply x 10^decimals`

Reply:
I changed this one to long long.

Reply:
what's the maximum could `amount` be in Oasis? is `long long` big enough?

Reply:
Done! I changed to uint256.